inluxclient/client.go

```diff
@@ -71,3 +91,83 @@ func New(params Params) (*Client, error) {
 	}
 	return c, nil
 }
+
+// makeAPICall issues an HTTP request to InfluxDB server API url according to parameters.
+// Additionally, sets Authorization header and User-Agent.
+// It returns http.response or Error
+func (c *Client) makeAPICall(ctx context.Context, params httpParams) (*http.Response, *ServerError) {
```

Comment:
I think I'd suggest returning `error` rather than `*ServerError`. Returning a non-interface type as an error runs a distinct risk of [nil-vs-non-nil error confusion](https://go.dev/doc/faq#nil_error), and I think it's nice for the caller to be able to distinguish between a local error (as returned in the error from `http.Client.Do`) and a server-returned error. We can document that the error will always be a `*ServerError` when we got a response from the server).

Reply:
Ok